src/test/java/mate/academy/dao/UserDaoImplTest.java

```diff
@@ -0,0 +1,95 @@
+package mate.academy.dao;
+
+import java.util.NoSuchElementException;
+import java.util.Set;
+import mate.academy.dao.impl.RoleDaoImpl;
+import mate.academy.dao.impl.UserDaoImpl;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class UserDaoImplTest extends AbstractTest {
+    private UserDao userDao;
+    private RoleDao roleDao;
+    private User user;
+    private Role role;
+
+    @Override
+    protected Class<?>[] entities() {
+        return new Class[]{User.class, Role.class};
+    }
+
+    @BeforeEach
+    void setup() {
+        user = new User();
+        user.setEmail("test@test.com");
+        user.setPassword("1234");
+        userDao = new UserDaoImpl(getSessionFactory());
+    }
+
+    @Test
+    void save_ok() {
+        User actual = userDao.save(user);
+        Assertions.assertNotNull(actual);
+        Assertions.assertEquals(1L, actual.getId());
+    }
+
+    @Test
+    void save_notOk() {
+        UserDao userDao1 = new UserDaoImpl(null);
+        Assertions.assertThrows(DataProcessingException.class, () ->
+                userDao1.save(user)
+        );
+    }
+
+    @Test
+    void findById_ok() {
+        User expected = userDao.save(user);
+        Long id = expected.getId();
+        User actual = userDao.findById(id).get();
+        Assertions.assertEquals(expected.getId(), actual.getId());
+        Assertions.assertEquals(expected.getEmail(), actual.getEmail());
+        Assertions.assertEquals(expected.getPassword(), actual.getPassword());
+    }
+
+    @Test
+    void findById_notOk() {
+        User expected = userDao.save(user);
+        Long id = expected.getId();
+        UserDao userDao1 = new UserDaoImpl(null);
```

Comment:
Instead of passing null as sessionFactory, consider using a mock object (e.g., using Mockito) to simulate the behavior of a sessionFactory that would cause a DataProcessingException to be thrown.
Please recheck all occurences